### PR TITLE
MTD reconstruction: code cleaning for old comments (issue #24452)

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

At the time of MTD TDR the reconstruction code was quickly integrated, due to the time constraints, leaving a number of minor comments pending for future cleaning. They were summarised in issue #24452 . Part of them was already addressed, this PR is meant to solve ~~most of~~ the items in the list https://github.com/cms-sw/cmssw/issues/24452#issuecomment-460488657 (mostly code aesthetics, by @kpedro88 @Dr15Jones @slava77 ). ~~Those not yet solved here will be separately addressed with the original reviewers in parallel to this PR review.~~

#### PR validation:

Code compiles and run, test wf 34634.0 provides the same DQM output.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
hold

waiting to fix residual issues

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
unhold

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
@smuzaffar @makortel the clang-tidy warnings for ```TrackTansformer``` look to me due to a protection in the code for possibly missing tokens. If that is removed the code should be clean already.

**Comment by silviodonato** [comment]
+operations

**Comment by srimanob** [comment]
+Upgrade

**Review by perrotta — COMMENTED** [review]
Just a couple of trivial possible adjustments...

File: RecoLocalFastTime/FTLRecProducers/plugins/MTDTrackingRecHitProducer.cc [hunk 1]
```diff
@@ -19,31 +19,18 @@
...
-  void run(edm::Handle<edmNew::DetSetVector<FTLCluster> > inputHandle, MTDTrackingDetSetVector& output);
+  void produce(edm::StreamID, edm::Event& evt, const edm::EventSetup& es) const override;
 
 private:
   const edm::EDGetTokenT<FTLClusterCollection> ftlbClusters_;  // collection of barrel digis
   const edm::EDGetTokenT<FTLClusterCollection> ftleClusters_;  // collection of endcap digis
 
-  const MTDGeometry* geom_;
-  const MTDClusterParameterEstimator* cpe_;
   edm::ESGetToken<MTDGeometry, MTDDigiGeometryRecord> mtdgeoToken_;
```
[the remark this answers is not in the document]
Comment (fabiocos) [reply RecoLocalFastTime/FTLRecProducers/plugins/MTDTrackingRecHitProducer.cc:34]: ok

File: RecoLocalFastTime/FTLRecProducers/plugins/MTDTrackingRecHitProducer.cc [hunk 2]
```diff
@@ -19,31 +19,18 @@
...
   edm::ESGetToken<MTDClusterParameterEstimator, MTDCPERecord> cpeToken_;
```
[the remark this answers is not in the document]
Comment (fabiocos) [reply RecoLocalFastTime/FTLRecProducers/plugins/MTDTrackingRecHitProducer.cc:35]: ok

**Comment by fabiocos** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical adjustments/updates
- Jenkins tests pass

**Comment by srimanob** [comment]
+Upgrade

Code cleaning with no changes expected.

**Comment by silviodonato** [comment]
+1

**Pull request merged**